src/battle.py

```diff
@@ -743,26 +894,62 @@ def handle_projectiles(self, owner_nation, proj, enemy, enemy_force, stats, enem
 
                             #Allow the troop being hit by the projectile to try and defend a little.
                             defense = p.target.get_ranged_defense(owner_nation.tech.get_best_in_category('material'))
+                            is_shield = p.target.weapons[1].shield 
+                            armor_pierce = p.launcher.get_ranged_weapon().armor_pierce
+                            siege_weapon = p.launcher.get_ranged_weapon().siege_weapon
+                            is_heavy = p.target.armor.heavy                     
+
+                            if is_shield:
+                                damage = int(damage/2)
+                               # print p.launcher.name + "'s ranged attack was deflected by "+p.target.name+"'s shield: "+str(damage)
+                            elif not is_heavy:
+                                damage *= 2
+                                #print p.launcher.name + "'s ranged attack was left unprotected by "+p.target.name+"'s lack of a shield: "+str(damage)
+
+                            if is_heavy:
+                                if armor_pierce == 1:
+                                    defense /= 2
+                                 #   print p.launcher.name + "'s ranged attack pierced  "+p.target.name+"'s heavy armor:  "+str(damage)
+                                elif armor_pierce == -1:
+                                    defense *= 2
+                                  #  print p.launcher.name + "'s ranged attack glanced off of "+p.target.name+"'s hevy armor: "+str(damage)
+                            else:
+                                if armor_pierce == 1:
+                                    defense *= 2
+                                   # print p.launcher.name + "'s ranged attack was cushioned by "+p.target.name+"'s light armor: "+str(damage)
+                                elif armor_pierce == -1:
+                                    defense /= 2
+                                    #print p.launcher.name + "'s ranged attack impacted "+p.target.name+"'s light armor: "+str(damage)
 
                             if random.randint(0, p.target.discipline) == 0:
                                 p.target.fatigue += 1
 
                             #Can't do less than 0 damage
                             total_damage = max(damage - defense, 0)
 
+                            original_health = p.target.health
+
                             p.target.health -= total_damage
 
                             if p.target.health <= 0:
-                                stats['troops_killed'] += 1
-                                stats[p.launcher.name]['kills'] += 1
-                                stats[p.launcher.name][weapon]['kills'] += 1
+                                overkill = 1
+                                if siege_weapon:
+                                    overkill += p.target.health*-1/original_health
```

Comment:
Not sure I fully understand the purpose of this line?

Reply:
Well I tried to implement a way to kill more than one unit at a time